packages/flutter_tools/lib/src/android/adb.dart

```diff
@@ -112,6 +118,22 @@ class Adb {
     return controller.stream;
   }
 
+  Future _populateDeviceNames(List<AdbDevice> devices) async {
+    for (AdbDevice device in devices) {
+      if (device.modelID == null) {
+        if (_idToNameCache[device.id] == null)
+          await _populateDeviceCache();
+        device.modelID = _idToNameCache[device.id];
```

Comment:
I have to admit I'm confused as to why this line is necessary. Isn't _idToNameCache[device.id] set from device.modelID in the first place (line 134 below)?